### PR TITLE
Make Token Label settable

### PR DESCRIPTION
**Pull request by simo5**
Refactor a bit how Token Info is managed and store some of the token data in a special object in the database.

Adjust test IDs to not stomp over the lower 10 IDs, consider them reserved for internal token objects.

### CODE REVIEW

File: src/pkcs11/interface.rs [hunk 1]
```diff
@@ -14,12 +14,16 @@ pub struct CK_INTERFACE {
...
-    CKR_VENDOR_DEFINED + KRYERR_OFFSET + 1;
+/* Object types */
+pub const KRY_TOKEN_DATA: CK_OBJECT_CLASS = KRY_VENDOR_OFFSET + 1;
+
+/* Attributes */
+pub const KRA_MAX_LOGIN_ATTEMPTS: CK_ATTRIBUTE_TYPE = KRY_VENDOR_OFFSET + 1;
+pub const KRA_FLAGS: CK_ATTRIBUTE_TYPE = KRY_VENDOR_OFFSET + 2;
+
+/* Errors */
+pub const KRYERR_TOKEN_NOT_INITIALIZED: CK_ULONG = KRY_VENDOR_OFFSET + 1;
```
Comment (Jakuje) [inline src/pkcs11/interface.rs:27]: should we stick to the three-letters prefix and do `KRR_` for return codes?

File: src/pkcs11/interface.rs [hunk 2]
```diff
@@ -14,12 +14,16 @@ pub struct CK_INTERFACE {
...
-pub const KRYATTR_OFFSET: CK_ULONG = 485259;
-pub const KRYATTR_MAX_LOGIN_ATTEMPTS: CK_ULONG =
-    CKA_VENDOR_DEFINED + KRYATTR_OFFSET + 1;
+pub const KRY_VENDOR_OFFSET: CK_ULONG = CKA_VENDOR_DEFINED + 485259;
 
-pub const KRYERR_OFFSET: CK_ULONG = 485259;
-pub const KRYERR_TOKEN_NOT_INITIALIZED: CK_ULONG =
-    CKR_VENDOR_DEFINED + KRYERR_OFFSET + 1;
+/* Object types */
+pub const KRY_TOKEN_DATA: CK_OBJECT_CLASS = KRY_VENDOR_OFFSET + 1;
```
Comment (Jakuje) [inline src/pkcs11/interface.rs:20]: Should we do the `KRO_` for object class?

The current way is slightly non-consistent.

**Review by Jakuje — APPROVED** [review]

File: src/pkcs11/interface.rs (hunk 1 above)
Comment (simo5) [reply src/pkcs11/interface.rs:27]: Had not thought about those, but yeah makes sense!

File: src/pkcs11/interface.rs (hunk 2 above)
Comment (simo5) [reply src/pkcs11/interface.rs:20]: Will change

**Pull request merged**